### PR TITLE
chore(tests): Moved priority queue tests from hardhat to foundry

### PR DESCRIPTION
**Pull request by mm-zk**
# What ❔

* Migrated priority queue tests from hardhat to foundry.
* Also added some more complex tests

## Why ❔

* Part of our larger test migration

image

### CODE REVIEW

File: l1-contracts/test/foundry/unit/concrete/PriorityQueue/_PriorityQueue_Shared.t.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {PriorityQueueTest} from "../../../../../cache/solpp-generated-contracts/dev-contracts/test/PriorityQueueTest.sol";
+import {PriorityOperation} from "../../../../../cache/solpp-generated-contracts/dev-contracts/test/PriorityQueueTest.sol";
```
Comment (koloz193) [inline l1-contracts/test/foundry/unit/concrete/PriorityQueue/_PriorityQueue_Shared.t.sol:6]: nit: can do `import {PriorityQueueTest, PriorityOperation} from "../../../../../cache/solpp-generated-contracts/dev-contracts/test/PriorityQueueTest.sol";`

**Review by koloz193 — APPROVED** [review]

**Review by StanislavBreadless — APPROVED** [review]
Agree with the comments from Zach. Otherwise, well done 👍

File: l1-contracts/test/foundry/unit/concrete/PriorityQueue/OnEmptyQueue.sol [hunk 2]
```diff
@@ -0,0 +1,22 @@
+pragma solidity 0.8.20;
+
+import {PriorityQueueSharedTest} from "./_PriorityQueue_Shared.t.sol";
+
+contract OnEmptyQueueTest is PriorityQueueSharedTest {
+    function test_gets() public {
+        assertEq(0, priorityQueue.getSize());
+        assertEq(0, priorityQueue.getFirstUnprocessedPriorityTx());
+        assertEq(0, priorityQueue.getTotalPriorityTxs());
+        assertTrue(priorityQueue.isEmpty());
```
Comment (vladbochok) [inline l1-contracts/test/foundry/unit/concrete/PriorityQueue/OnEmptyQueue.sol:10]:
```suggestion
        assertEq(0, priorityQueue.getPriorityQueueSize());
        assertTrue(priorityQueue.isEmpty());
```

**Review by vladbochok — APPROVED** [review]

File: l1-contracts/test/foundry/unit/concrete/PriorityQueue/_PriorityQueue_Shared.t.sol (hunk 1 above)
Comment (benceharomi) [reply l1-contracts/test/foundry/unit/concrete/PriorityQueue/_PriorityQueue_Shared.t.sol:6]: You can also shorten imports:
```
import {PriorityQueueTest, PriorityOperation} from "solpp/dev-contracts/test/PriorityQueueTest.sol";
```

File: l1-contracts/test/foundry/unit/concrete/PriorityQueue/OnEmptyQueue.sol (hunk 2 above)
Comment (mm-zk) [reply l1-contracts/test/foundry/unit/concrete/PriorityQueue/OnEmptyQueue.sol:10]: but priorityQueue has only 'getSize' method - that I check 3 lines above, and it doesn't have getPriorityQueueSize.
or am I missing something?

Comment (vladbochok) [reply l1-contracts/test/foundry/unit/concrete/PriorityQueue/OnEmptyQueue.sol:10]: You are right, ignore my comment above.

**Review by benceharomi — APPROVED** [review]

**Pull request merged**